src/jsifier.js

```diff
@@ -1,549 +1,561 @@
+      } else if (typeof snippet === 'function') {
+        isFunction = true;
+        snippet = processLibraryFunction(snippet, ident, finalName);
+        Functions.libraryFunctions[finalName] = 1;
+      }
+
+      var postsetId = ident + '__postset';
+      var postset = LibraryManager.library[postsetId];
+      if (postset) {
+        // A postset is either code to run right now, or some text we should emit.
+        // If it's code, it may return some text to emit as well.
+        if (typeof postset === 'function') {
+          postset = postset();
+        }
+        if (postset && !addedLibraryItems[postsetId] && !SIDE_MODULE) {
+          addedLibraryItems[postsetId] = true;
+          itemsDict.GlobalVariablePostSet.push({
+            JS: postset + ';'
+          });
+        }
+      }
+
+      if (redirectedIdent) {
+        deps = deps.concat(LibraryManager.library[redirectedIdent + '__deps'] || []);
+      }
+      // In asm, dependencies implemented in C might be needed by JS library functions.
+      // We don't know yet if they are implemented in C or not. To be safe, export such
+      // special cases.
+      [LIBRARY_DEPS_TO_AUTOEXPORT].forEach(function(special) {
+        deps.forEach(function(dep) {
+          if (dep == special && !EXPORTED_FUNCTIONS[dep]) {
+            EXPORTED_FUNCTIONS[dep] = 1;
+          }
+        });
+      });
+      if (VERBOSE) printErr('adding ' + finalName + ' and deps ' + deps + ' : ' + (snippet + '').substr(0, 40));
+      var depsText = (deps ? '\n' + deps.map(addFromLibrary).filter(function(x) { return x != '' }).join('\n') : '');
+      var contentText;
+      if (isFunction) {
+        // Emit the body of a JS library function.
+        var proxyingMode = LibraryManager.library[ident + '__proxy'];
+        if (USE_PTHREADS && proxyingMode) {
+          if (proxyingMode !== 'sync' && proxyingMode !== 'async') {
+            throw 'Invalid proxyingMode ' + ident + '__proxy: \'' + proxyingMode + '\' specified!';
+          }
+          var sync = proxyingMode === 'sync';
+          assert(typeof original === 'function');
+          contentText = modifyFunction(snippet, function(name, args, body) {
+            return 'function ' + name + '(' + args + ') {\n' +
+                   'if (ENVIRONMENT_IS_PTHREAD) return _emscripten_proxy_to_main_thread_js(' + proxiedFunctionTable.length + ', ' + (+sync) + (args ? ', ' : '') + args + ');\n' + body + '}\n';
+          });
+          proxiedFunctionTable.push(finalName);
+        } else {
+          contentText = snippet; // Regular JS function that will be executed in the context of the calling thread.
+        }
+      } else if (typeof snippet === 'string' && snippet.indexOf(';') == 0) {
+        contentText = 'var ' + finalName + snippet;
+        if (snippet[snippet.length-1] != ';' && snippet[snippet.length-1] != '}') contentText += ';';
+      } else if (typeof snippet === 'undefined') {
+        contentText = '';

```

Comment:
Why is this change needed?

Reply:
Without this, the code will go down to the block below and we end up with
`var sidey=undefined;`